ui/app/hooks/useCancelTransaction.js

```diff
@@ -19,27 +25,60 @@ import { getConversionRate, getSelectedAccount } from '../selectors';
  * @return {[boolean, Function]}
  */
 export function useCancelTransaction(transactionGroup) {
-  const { primaryTransaction, initialTransaction } = transactionGroup;
+  const { primaryTransaction } = transactionGroup;
   const gasPrice = primaryTransaction.txParams?.gasPrice?.startsWith('-')
     ? '0x0'
     : primaryTransaction.txParams?.gasPrice;
-  const { id } = initialTransaction;
+  const transaction = primaryTransaction;
   const dispatch = useDispatch();
   const selectedAccount = useSelector(getSelectedAccount);
   const conversionRate = useSelector(getConversionRate);
+  const defaultNewGasPrice = addHexPrefix(
+    multiplyCurrencies(gasPrice, 1.2, {
+      toNumericBase: 'hex',
+      multiplicandBase: 16,
+      multiplierBase: 10,
+    }),
+  );
+
   const cancelTransaction = useCallback(
     (event) => {
       event.stopPropagation();
-
+      dispatch(setCustomGasLimit('0x5208'));
+      dispatch(setCustomGasPriceForRetry(defaultNewGasPrice));
+      const tx = {
+        ...transaction,
+        txParams: {
+          ...transaction.txParams,
+          gas: '0x5208',
+          value: '0x0',
+        },
+      };
       return dispatch(
-        showModal({
-          name: 'CANCEL_TRANSACTION',
-          transactionId: id,
-          originalGasPrice: gasPrice,
+        showSidebar({
+          transitionName: 'sidebar-left',
+          type: 'customize-gas',
+          props: {
+            transaction: tx,
+            onSubmit: (gl, gp) => {
+              const userCustomizedGasTotal = getHexGasTotal({
+                gasPrice: gp,
+                gasLimit: gl || '0x5208',
```

Comment:
Is it possible for `gl` to be falsey here? It didn't seem possible when I traced it down, because this value comes from `customModalGasLimitInHex` which is always set to something (it has a fallback of `0x5208` in `gas-modal-page-container.container.js`).

Reply:
could we use `newGasLimiit`/`newGasPrice` here for readability?